### PR TITLE
Add engine support

### PR DESCRIPTION
**Pull request by tmquinn**
There's a couple skipped tests in here right now. We should probably discuss if we want to support those features:
- errors if lookup of a route-map does not specify isRouteMap
- will return the raw value if no 'default' is available

### CODE REVIEW

File: tests/unit/strict-resolver/basic-test.js [hunk 1]
```diff
@@ -135,3 +137,148 @@ test("can lookup a view in another namespace with different syntax", function(as
   assert.ok(view, 'view was returned');
   assert.equal(view, expected, 'default export was returned');
 });
+
+test("can lookup a view", function(assert) {
```
Comment (rwjblue) [inline tests/unit/strict-resolver/basic-test.js]: 🔥 views are no longer a thing 😈

Comment (tmquinn) [inline tests/unit/strict-resolver/basic-test.js]: Mostly just adding tests from ember-resolver as a starting point. Getting to the point where each new test brings up design questions.

Comment (rwjblue) [inline tests/unit/strict-resolver/basic-test.js]: Yeah, I figured.  Seems like we should have started by forking that repo.  Its _slightly_ annoying to loose all the proper attributions and bug history / reasoning.

File: tests/unit/strict-resolver/basic-test.js [hunk 2]
```diff
@@ -104,34 +106,131 @@ test("can lookup something in another namespace with different syntax", function
...
+    if (!test) {
+      assert.equal(message, 'Attempted to lookup "helper:reverseList". Use "helper:reverse-list" instead.');
+    }
+  };
+
+  var helper = resolver.resolve('helper:reverseList');
+  assert.ok(!helper, 'no helper was returned');
+});
+
+skip('errors if lookup of a route-map does not specify isRouteMap', function(assert) {
```
Comment (stefanpenner) [inline tests/unit/strict-resolver/basic-test.js]: should we keep skipped tests around

Comment (tmquinn) [inline tests/unit/strict-resolver/basic-test.js]: Oh, I rewote the description and left this out. So the skipped tests are basically questions about wether that is behavior we want to support... was looking for your feedback on that...

Comment (stefanpenner) [inline tests/unit/strict-resolver/basic-test.js]: i think we should delete the skipped tests, if we don't intend to make them work in the next little while.

Then we can ship.

Comment (stefanpenner) [inline tests/unit/strict-resolver/basic-test.js]: we can delete this test

File: tests/unit/strict-resolver/basic-test.js [hunk 3]
```diff
@@ -104,34 +106,131 @@ test("can lookup something in another namespace with different syntax", function
...
     return { default: expected };
   });
 
-  var view = resolver.resolve('view:other@post');
+  assert.throws(() => {
+    resolver.resolve('route-map:foo-bar');
+  }, /The route map for foo-bar should be wrapped by 'buildRoutes' before exporting/);
+});
+
+skip("will return the raw value if no 'default' is available", function(assert) {
```
Comment (stefanpenner) [inline tests/unit/strict-resolver/basic-test.js]: I believe we can delete this test

**Comment by stefanpenner** [comment]
released as v0.2.0 🎉

**Pull request merged**